extensions/stackdriver/stackdriver.cc

```diff
@@ -141,38 +142,19 @@ void StackdriverContext::onLog() {
   bool isOutbound = getRootContext()->isOutbound();
   ::Wasm::Common::populateHTTPRequestInfo(isOutbound, &request_info_);
 
+  auto key = isOutbound ? kUpstreamMetadataKey : kDownstreamMetadataKey;
+
   // Fill in peer node metadata in request info.
-  if (isOutbound) {
-    google::protobuf::Struct upstream_metadata;
-    if (getMetadataStruct(Common::Wasm::MetadataType::Request,
-                          kUpstreamMetadataKey,
-                          &upstream_metadata) != Common::Wasm::WasmResult::Ok) {
-      logWarn(absl::StrCat("cannot get metadata for: ", kUpstreamMetadataKey));
-      return;
-    }
-
-    auto status = ::Wasm::Common::extractNodeMetadata(upstream_metadata,
-                                                      &peer_node_info_);
-    if (status != Status::OK) {
-      logWarn("cannot parse upstream peer node metadata " +
-              upstream_metadata.DebugString() + ": " + status.ToString());
-    }
-  } else {
-    google::protobuf::Struct downstream_metadata;
-    if (getMetadataStruct(Common::Wasm::MetadataType::Request,
-                          kDownstreamMetadataKey, &downstream_metadata) !=
-        Common::Wasm::WasmResult::Ok) {
-      logWarn(
-          absl::StrCat("cannot get metadata for: ", kDownstreamMetadataKey));
-      return;
-    }
-
-    auto status = ::Wasm::Common::extractNodeMetadata(downstream_metadata,
-                                                      &peer_node_info_);
-    if (status != Status::OK) {
-      logWarn("cannot parse downstream peer node metadata " +
-              downstream_metadata.DebugString() + ": " + status.ToString());
-    }
+  google::protobuf::Value metadata;
```

Comment:
This should be wrapper of macro than this often used magic string - “filter_state”

Reply:
Are we committed to "filter_state"? I didn't want to introduce API changes before we're sure about the naming. Same applies to `traffic_direction`.